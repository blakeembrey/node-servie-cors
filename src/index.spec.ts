import { compose } from "throwback";
import { cors } from "./index";
import { Request, Response } from "servie/dist/node";
import { finalhandler } from "servie-finalhandler";

describe("servie-cors", () => {
  it("should set origin", () => {
    const app = compose<Request, Response>([
      cors(),
      req => new Response(req.method, { status: 200 })
    ]);

    const req = new Request("/");

    return app(req, finalhandler(req)).then(res => {
      expect(res).toMatchSnapshot();
    });
  });

  it("should handle options", () => {
    const app = cors();

    const req = new Request("/", { method: "options" });

    return app(req, finalhandler(req)).then(res => {
      expect(res).toMatchSnapshot();
    });
  });

  it("should forward options", () => {
    const app = compose<Request, Response>([
      cors({ optionsContinue: true }),
      function(req) {
        return new Response(req.method, { status: 200 });
      }
    ]);

    const req = new Request("/", { method: "options" });

    return app(req, finalhandler(req)).then(res => {
      expect(res).toMatchSnapshot();
    });
  });
});
