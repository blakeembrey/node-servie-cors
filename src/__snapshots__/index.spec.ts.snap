// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`servie-cors should forward options 1`] = `
Response {
  "$rawBody": "options",
  "headers": Headers {
    "object": Object {
      "access-control-allow-methods": "GET,HEAD,PUT,PATCH,POST,DELETE",
      "access-control-allow-origin": "*",
      "content-length": "7",
      "content-type": "text/plain",
    },
  },
  "status": 200,
  "statusText": "",
  "trailer": Promise {},
}
`;

exports[`servie-cors should handle options 1`] = `
Response {
  "$rawBody": null,
  "headers": Headers {
    "object": Object {
      "access-control-allow-methods": "GET,HEAD,PUT,PATCH,POST,DELETE",
      "access-control-allow-origin": "*",
    },
  },
  "status": 204,
  "statusText": "",
  "trailer": Promise {},
}
`;

exports[`servie-cors should set origin 1`] = `
Response {
  "$rawBody": "GET",
  "headers": Headers {
    "object": Object {
      "access-control-allow-origin": "*",
      "content-length": "3",
      "content-type": "text/plain",
    },
  },
  "status": 200,
  "statusText": "",
  "trailer": Promise {},
}
`;
